cmd/webhook/app/webhook.go

```diff
@@ -83,41 +89,177 @@ func NewServerWithOptions(log logr.Logger, opts options.WebhookOptions) (*server
 	}
 
 	return &server.Server{
-		ListenAddr:        fmt.Sprintf(":%d", opts.ListenPort),
-		HealthzAddr:       fmt.Sprintf(":%d", opts.HealthzPort),
-		PprofAddr:         opts.PprofAddress,
+		ListenAddr:        fmt.Sprintf(":%d", *opts.SecurePort),
+		HealthzAddr:       fmt.Sprintf(":%d", *opts.HealthzPort),
 		EnablePprof:       opts.EnablePprof,
+		PprofAddr:         opts.PprofAddress,
 		CertificateSource: source,
-		CipherSuites:      opts.TLSCipherSuites,
-		MinTLSVersion:     opts.MinTLSVersion,
+		CipherSuites:      opts.TLSConfig.CipherSuites,
+		MinTLSVersion:     opts.TLSConfig.MinTLSVersion,
 		ValidationWebhook: validationHook,
 		MutationWebhook:   mutationHook,
 		ConversionWebhook: conversionHook,
 		Log:               log,
 	}, nil
 }
 
+const componentWebhook = "webhook"
+
 func NewServerCommand(stopCh <-chan struct{}) *cobra.Command {
-	var opts options.WebhookOptions
+	ctx := cmdutil.ContextWithStopCh(context.Background(), stopCh)
+	ctx = logf.NewContext(ctx, nil, "webhook")
+	log := logf.FromContext(ctx)
+
+	cleanFlagSet := pflag.NewFlagSet(componentWebhook, pflag.ContinueOnError)
+	// Replaces all instances of `_` in flag names with `-`
+	cleanFlagSet.SetNormalizeFunc(cliflag.WordSepNormalizeFunc)
+	webhookFlags := options.NewWebhookFlags()
+	webhookConfig, err := options.NewWebhookConfiguration()
+	if err != nil {
+		log.Error(err, "Failed to create new webhook configuration")
+		os.Exit(1)
+	}
 
 	cmd := &cobra.Command{
-		Use:   "webhook",
-		Short: fmt.Sprintf("Webhook component providing API validation, mutation and conversion functionality for cert-manager (%s) (%s)", util.AppVersion, util.AppGitCommit),
-		RunE: func(cmd *cobra.Command, args []string) error {
-			ctx := cmdutil.ContextWithStopCh(context.Background(), stopCh)
-			ctx = logf.NewContext(ctx, nil, "webhook")
-			log := logf.FromContext(ctx)
-
-			srv, err := NewServerWithOptions(log, opts)
+		Use:  componentWebhook,
+		Long: fmt.Sprintf("Webhook component providing API validation, mutation and conversion functionality for cert-manager (%s) (%s)", util.AppVersion, util.AppGitCommit),
```

Comment:
In upstream components, only `Long` is set - in this PR we also explicitly specify the 'help format' text too, and the `Short` text isn't included:

```
	cmd.SetHelpFunc(func(cmd *cobra.Command, args []string) {
		fmt.Fprintf(cmd.OutOrStdout(), "%s\n\n"+usageFmt, cmd.Long, cmd.UseLine(), cleanFlagSet.FlagUsagesWrapped(2))
	})
```

So all in all, it doesn't actually make any difference. I just followed the upstream pattern and used `Long` because if we may well also end out with longer help text here anyway (considering this is the only text being shown) 🤷